velox/functions/prestosql/window/tests/WindowTestBase.h

```diff
@@ -13,43 +13,46 @@
 
 namespace facebook::velox::window::test {
 
-/// Common set of window function over clauses using a combination of three
-/// columns.
-static std::vector<std::string> kBasicOverClauses = {
-    "partition by c0 order by c1 desc, c2",
-    "partition by c1 order by c0, c2 desc",
-    "partition by c0 order by c2, c1",
-    "partition by c1 order by c2, c0 desc",
-    // No partition by clause.
-    "order by c0 asc nulls first, c1 desc, c2",
-    "order by c1 asc, c0 desc nulls last, c2 desc",
-    "order by c0 asc, c2 desc, c1 asc nulls last",
-    "order by c2 asc, c1 desc nulls first, c0",
-    // No order by clause.
-    "partition by c0, c1, c2",
-};
+/// Exhaustive set of window function over clauses using a combination of four
+/// columns. Columns c0 and c1 have input data meant to test partitioning
+/// and sorting behavior of the operator. Columns c2 and c3 are used with
+/// frame clauses. The individual functions might add new columns if required.
+static std::vector<std::string> kOverClauses = {
+    "partition by c0 order by c1 desc, c2, c3",
+    "partition by c1 order by c0, c2 desc, c3",
+    "partition by c0 order by c2, c1, c3",
+    "partition by c1 order by c2, c0 desc, c3",
+
+    // Order by with asc/desc and nulls first/last.
+    "partition by c0 order by c1 nulls first, c2, c3",
+    "partition by c0, c2 order by c1 nulls first, c3",
+    "partition by c1 order by c0 nulls first, c2, c3",
+    "partition by c1, c2 order by c0 nulls first, c3",
+    "partition by c0 order by c1 desc nulls first, c2, c3",
+    "partition by c0, c2 order by c1 desc nulls first, c3",
+    "partition by c1 order by c0 desc nulls first, c2, c3",
+    "partition by c1, c2 order by c0 desc nulls first, c3",
 
-/// Common set of window function over clauses with different sort orders
-/// using a combination of three columns.
-static std::vector<std::string> kSortOrderBasedOverClauses = {
-    "partition by c0 order by c1 nulls first, c2",
-    "partition by c0, c2 order by c1 nulls first",
-    "partition by c1 order by c0 nulls first, c2",
-    "partition by c1, c2 order by c0 nulls first",
-    "partition by c0 order by c1 desc nulls first, c2",
-    "partition by c0, c2 order by c1 desc nulls first",
-    "partition by c1 order by c0 desc nulls first, c2",
-    "partition by c1, c2 order by c0 desc nulls first",
     // No partition by clause.
-    "order by c0 asc nulls first, c1 desc nulls first, c2",
-    "order by c1 asc nulls first, c0 desc nulls first, c2",
-    "order by c0 desc nulls first, c1 asc nulls first, c2",
-    "order by c1 desc nulls first, c0 asc nulls first, c2",
+    "order by c0 asc nulls first, c1 desc, c2, c3",
+    "order by c1 asc, c0 desc nulls last, c2 desc, c3",
+    "order by c0 asc, c2 desc, c1 asc nulls last, c3",
+    "order by c2 asc, c1 desc nulls first, c0, c3",
+
+    // No partition by clause, order by with asc/desc nulls first/last.
```

Comment:
This comment suggest that the clauses below have "asc/desc nulls first/last" and implies that clauses before don't have these, but I see both sets of clauses having "asc/desc and nulls". Would you clarify?

Reply:
Thanks for pointing this out. This comment is not needed anymore.